src/QMCHamiltonians/tests/test_paircorr.cpp

```diff
@@ -0,0 +1,197 @@
+  bool lat_okay = doc.parseFromString(lat_xml);
+  REQUIRE(lat_okay);
+  xmlNodePtr lat_xml_root = doc.getRoot();
+
+  ParticleSetPool pset_builder(c, "pset_builder");
+  pset_builder.putLattice(lat_xml_root); // Builds lattice
+
+  bool pset_okay = doc.parseFromString(pset_xml);
+  REQUIRE(pset_okay);
+  xmlNodePtr pset_xml_root = doc.getRoot();
+  pset_builder.put(pset_xml_root);  // Builds ParticleSet
+
+  // Get the (now assembled) ParticleSet, do simple sanity checks, then print info 
+  ParticleSet* elec = pset_builder.getParticleSet("e");
+  elec->Lattice.copy(Lattice); // copy in the new Lattice
+
+  REQUIRE(elec->SameMass);
+  REQUIRE(elec->getName() == "e");
+
+  // Move the particles manually onto B1 lattice
+  // NB: Spins are grouped contiguously
+  // Up spins
+  elec->R[0][0] = 0.0;
+  elec->R[0][1] = 0.0;
+  elec->R[0][2] = 0.0;
+  elec->R[1][0] = 1.0;
+  elec->R[1][1] = 1.0;
+  elec->R[1][2] = 0.0;
+  elec->R[2][0] = 1.0;
+  elec->R[2][1] = 0.0;
+  elec->R[2][2] = 1.0;
+  elec->R[3][0] = 0.0;
+  elec->R[3][1] = 1.0;
+  elec->R[3][2] = 1.0;
+
+  // Down spins
+  elec->R[4][0] = 1.0;
+  elec->R[4][1] = 0.0;
+  elec->R[4][2] = 0.0;
+  elec->R[5][0] = 0.0;
+  elec->R[5][1] = 1.0;
+  elec->R[5][2] = 0.0;
+  elec->R[6][0] = 0.0;
+  elec->R[6][1] = 0.0;
+  elec->R[6][2] = 1.0;
+  elec->R[7][0] = 1.0;
+  elec->R[7][1] = 1.0;
+  elec->R[7][2] = 1.0;
+
+  elec->get(std::cout); // print particleset info to stdout
+
+  // Set up the distance table
+  const int ee_table_id = elec->addTable(*elec, DT_AOS, false);
+  const DistanceTableData& dii(elec->getDistTable(ee_table_id));
+  elec->update(); // distance table evaluation here
+
+  // Make a PairCorrEstimator, call put() to set up internals
+  std::string name = elec->getName();
+  PairCorrEstimator* paircorr = new PairCorrEstimator(*elec, name);
+  bool gofr_okay = doc.parseFromString(gofr_xml);
+  REQUIRE(gofr_okay);
+  xmlNodePtr gofr_xml_root = doc.getRoot();
+  paircorr->put(gofr_xml_root);
+  paircorr->addObservables( elec->PropertyList, elec->Collectables );
+
+  // Compute g(r) then print it to stdout
+  // NB: Hardcoded to match hardcoded xml above. ***Fragile!!!***
+  paircorr->evaluate(*elec);
+
+  auto gofr = elec->Collectables;
+  std::cout << "\n";
+  std::cout << "gofr:\n";
+  std::cout << std::fixed;
+  std::cout << std::setprecision(6);
+  std::cout <<  std::setw(4) << "i" 
+	    << "  " << std::setw(12) << "r" 
+	    << "  " << std::setw(12) << "uu" 
+	    << "  " << std::setw(12) << "ud" 
+	    << "  " << std::setw(12) << "dd" << "\n";
+  std::cout << "============================================================\n";
+  for ( int i(0); i<20; i++ )
+    {
+      std::cout << std::setw(4) << i
+		<< "  " << std::setw(12) << i*0.10
+  		<< "  " << std::setw(12) << gofr[i]
+  		<< "  " << std::setw(12) << gofr[i+20]
+  		<< "  " << std::setw(12) << gofr[i+40] 
+  		<< "\n";
+    } 
+
+  // Verify against analytic result.
+  const RealType eps = 1E-01;  // tolerance
```

Comment:
Is there are reason for this large tolerance (absolute difference for checks)? Would you be happy to make it stricter?

Reply:
The tolerance is small because about 0.05-0.01 disagreement between analytic and QMCPACK result, depending on the gofr bin. I'm not sure where the source of disagreement is coming from, as the norm_factor is now exact.

With the fix from Mark and the changes to norm_factor, a test on a more realistic system now gives expected results with the expected normalization. 256 electron HEG with rs=1.0 no jastrow (SOA + fixes):
![gofr](https://user-images.githubusercontent.com/25334555/68043937-0fb30900-fc9c-11e9-9651-0d368b306dbc.png)

Reply:
Same problem before fixes:
![gofr old](https://user-images.githubusercontent.com/25334555/68044006-36713f80-fc9c-11e9-9523-c798f4d3aa02.png)